test/Microsoft.AspNetCore.SignalR.Test.Server/EchoEndPoint.cs

```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.IO.Pipelines;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Sockets;
+
+namespace Microsoft.AspNetCore.SignalR.Test.Server
+{
+    public class EchoEndPoint : EndPoint
+    {
+        public async override Task OnConnectedAsync(Connection connection)
+        {
+            var buffer = new byte[256];
+            var bytesRead = 0;
+
+            while ((bytesRead = await connection.Channel.Input.ReadAsync(new Span<byte>(buffer))) > 0)
```

Comment:
OK. Will update. This is a new API and because of the extension methods using it the way it is supposed to be used is confusing.